chainstate/src/interface/chainstate_interface_impl.rs

```diff
@@ -94,9 +94,8 @@ impl<S: BlockchainStorage, V: TransactionVerificationStrategy> ChainstateInterfa
         self.chainstate
             .query()
             .map_err(ChainstateError::from)?
-            .get_block_height_in_main_chain(&(*block_id).into())
+            .is_block_in_main_chain(&(*block_id).into())
             .map_err(ChainstateError::FailedToReadProperty)
```

Comment:
Note that `ChainstateRef::is_block_in_main_chain` takes `&Id<GenBlock>`, but this one takes `&Id<Block>`, which is a bit strange. I wasn't sure what to use in `ChainstateQuery::is_block_in_main_chain` but decided to go with `Id<GenBlock>` in the end.
It would be nice to unify their interfaces though. Should I do it? Which kind of block id would be more logical in this case?

Reply:
Sure... though Genesis is always part of main chain, but it's convenient to just be able to submit a id of genblock.

Reply:
Ok, I've changed `ChainstateInterface::is_block_in_main_chain`, so that it uses `&Id<GenBlock>` too.